wee_slack.py

```diff
@@ -1422,16 +1423,46 @@ def update_member_presence(self, user, presence):
                 c.update_nicklist(user.id)
 
     def subscribe_users_presence(self):
-        # FIXME: There is a limitation in the API to the size of the
-        # json we can send.
-        # We should try to be smarter to fetch the users whom we want to
-        # subscribe to.
-        users = list(self.users.keys())[0:750]
+        if len(self.users) <= PRESENCE_SUBSCRIBE_SIZE:
+            users = list(self.users.keys())
+        else:
+            users = self.get_users_by_presence_order()
+
         self.send_to_websocket({
             "type": "presence_sub",
             "ids": users,
         }, expect_reply=False)
 
+    def get_users_by_presence_order(self):
+        """
+        Get users in this order (restricted by the API JSON size limit):
+
+        1. Current user
+        2. Users in DMs
+        3. Users in MPDMs
+        4. Users in private groups
+        5. Users in public channels (excluding the "general" channel)
+        """
+        users = {
+            channel_type: set() for channel_type in
+            ("im", "mpim", "group", "channel", "shared")
+        }
+        for channel in self.channels.values():
+            if (
+                getattr(channel, "is_open", False)
+                or getattr(channel, "is_member", False)
```

Comment:
`and` has a higher presence than `or`, so you need parenthesis around the first two conditions.

Reply:
I think it's probably more sensible to only check `is_general` if it might be a public channel. I say it's mostly fine as is, but maybe add parens anyway to make the intent clear to a casual reader.

Reply:
Only public channels have `is_member` and `is_general` so I agree it makes sense to group those together instead. It should also be fine to assume `is_general` exists if `is_member` exists so `getattr(channel, "is_general", False)` isn't necessary.